### ISSUE TITLE
Weapons chambered for 10mm auto should adhere to game_balance.md

### ISSUE THREAD

**Issue by RenechCDDA** [CleverRaven/Cataclysm-DDA#49369]
# Is your feature request related to a problem? Please describe.
10mm auto is a pistol cartridge. Despite this, the common pistols chambered for it have staggering damage penalties. For example, the glock 20 has a -6(!!) damage penalty despite having a barrel length of 4.61 inches*. This is longer than the 4 inch normal barrel length stated in game_balance.md, so according to its formula the damage bonus should be +0. A -6 penalty implies that the barrel length is actually negative.... somehow.

*Quick google search found glock's website stating 4.61in https://us.glock.com/en/pistols/g20-gen4

This applies to all weapons chambered in 10mm auto. The MP5/10, a submachine gun(!) firing a pistol round, still has a -1 damage penalty.

# Describe the solution you'd like
Update game_balance.md? (Maybe 10mm auto is supposed to have a damage penalty across the board? It's not unprecedented.)

Buff the weapons?

# Describe alternatives you've considered
Audit all firearms for adherence?

# Additional context
These were added in #42636 and it seems the same formula currently in game_balance.md was used at that time. As such it's unclear to me how the damage modifiers were decided on in this PR; they do not follow the game_balance.md and even the presented data seems to have errors. e.g. The MP5/10 should have +1 according to the table, but actually has -1 as committed.

~~I'd be happy to submit a PR adjusting the damage values on a per-gun basis, but I worry that some of the other values (volume) might be bunk too and I don't fully know the calculations behind that.~~ A PR was kindly submitted by another contributor, and I've learned that the volume requirements should be *okay*. They were generated using an older method, so ideally they should get updated at some point but there's no reason to believe they are wrong.

**Comment by RenechCDDA** [comment]
I did some sleuthing of commit histories, mostly to check when/if balance ideals would have changed. As far as the history of game_balance.md goes, the +- 3inches of barrel length per point of damage rule extends back to its inception on 23 November 2017. 

However, I did find an enlightening reference in #37691, which seems to suggest the author was using a novel and undocumented calculation for muzzle velocity to translate into damage. I am not able to find the spreadsheet they refer to, but I assume the "shotguns PR" is 34638. So.... this issue definitely seems to extend past *only* 10mm firearms.

Funny thing though, (12 gauge) shot does not appear to be listed in game_balance.md. So technically the weird place shotguns are in is OK according to current ideal implementation. That said, **game_balance.md needs to be updated to add them**, or otherwise some indication needs to be given if shotguns are being special-cased. If not, I'm happy to take a look over them and modify damages accordingly. Really looking more and more like an entire firearm audit should happen.

Tentative ping: @mlangsdorf ?
